### ISSUE TITLE
accepting state checkpoints for all modes

### ISSUE THREAD

**Issue by Vatsal-Malaviya** [kerner-lab/Mars-Bench#25]
Saved weights with implemented models should be accepted.

Requirement: In training mode, the type of training should be followed even when using state checkpoints.

known bug: no warning for pre-trained: false mode when state_dict shared by user
